c10/cuda/CUDAMathCompat.h

```diff
@@ -149,6 +149,13 @@ __MATH_FUNCTIONS_DECL__ double normcdf(double x) {
   return ::normcdf(x);
 }
 
+__MATH_FUNCTIONS_DECL__ float exp(c10::complex<float> x) {
```

Comment:
why do we need these? We already support `exp` on complex numbers, and return type here is wrong.

Reply:
There was a comment from another reviewer and I made these changes. I will remove them if they are not necessary.

https://github.com/pytorch/pytorch/pull/89649#discussion_r1033671651

Reply:
They should not be necessary.

Reply:
Take a look at how `exp` is currently implemented.